src/libraries/System.Private.CoreLib/src/System/Threading/PortableThreadPool.cs

```diff
@@ -314,7 +314,7 @@ private void NotifyWorkItemProgress(object threadLocalCompletionCountObject, int
             ThreadInt64PersistentCounter.Increment(threadLocalCompletionCountObject);
             _separated.lastDequeueTime = currentTimeMs;
 
-            if (ShouldAdjustMaxWorkersActive(currentTimeMs))
+            if (!HillClimbing.IsDisabled && ShouldAdjustMaxWorkersActive(currentTimeMs))
```

Comment:
I just moved this check since I noticed that `ShouldAdjustMaxWorkersActive` doesn't get inlined